<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>812</width>
    <height>542</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="styleSheet">
    <string notr="true"/>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QComboBox" name="m_engine_box">
        <item>
         <property name="text">
          <string>César</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Vigenère</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Enigma</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Lorenz SZ40</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="m_cipher_mode">
        <item>
         <property name="text">
          <string>Cipher</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Decipher</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="m_reset_engine">
        <property name="text">
         <string>Reset Engine</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QTextEdit" name="m_text">
        <property name="styleSheet">
         <string notr="true"/>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QTextEdit" name="m_new_text">
        <property name="styleSheet">
         <string notr="true"/>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QWidget" name="widget" native="true">
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <layout class="QVBoxLayout" name="m_engine_layout"/>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QFrame" name="frame">
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QCheckBox" name="m_view_distribution">
         <property name="text">
          <string>View Letter Distribution</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="m_distribution_layout"/>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>812</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>m_text</sender>
   <signal>textChanged()</signal>
   <receiver>MainWindow</receiver>
   <slot>setCipheredText()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>250</x>
     <y>249</y>
    </hint>
    <hint type="destinationlabel">
     <x>405</x>
     <y>195</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_engine_box</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>setCipherEngine(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>405</x>
     <y>60</y>
    </hint>
    <hint type="destinationlabel">
     <x>405</x>
     <y>195</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_cipher_mode</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>changeCipherMode()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>658</x>
     <y>61</y>
    </hint>
    <hint type="destinationlabel">
     <x>405</x>
     <y>195</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_view_distribution</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>hideDistributionWidgets(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>405</x>
     <y>481</y>
    </hint>
    <hint type="destinationlabel">
     <x>405</x>
     <y>270</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>setCipheredText()</slot>
  <slot>setCipherEngine(int)</slot>
  <slot>changeCipherMode()</slot>
  <slot>hideDistributionWidgets(bool)</slot>
 </slots>
</ui>
